src/code/client.coffee

```diff
@@ -229,6 +230,12 @@ class CloudFileManagerClient
             return @alert(err) if err
             @_fileOpened content, metadata, {openedContent: content.clone()}, @_getHashParams metadata
 
+  openUrlFile: (url) ->
+    urlProvider = new URLProvider()
```

Comment:
Instead of instantiating it here for each call you could add it to the list of providers that are setup at startup on line 38.